### PR TITLE
Update Source.java and associate files to include more fields

### PR DESCRIPTION
**Pull request by DoItTomorrow**
Source.java now has a compulsory author field.
Source.java can now accept BiblioFields on construction.
Current implementation is by overloading Source constructor.

#75 #123

### CODE REVIEW

**Review by suriruhani — COMMENTED** [review]
Can you please specify what commands are affected by this update? For example I now see `search` would need to be able to take in and search by an `author` argument.

@DoItTomorrow

File: src/main/java/seedu/address/model/source/Source.java [hunk 1]
```diff
@@ -21,17 +22,31 @@
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
     private final Title title;
+    private final Author author;
     private final Type type;
     private final Detail detail;
 
     private final Set<Tag> tags = new HashSet<>();
+    private final Set<BiblioField> biblioFields = new HashSet<>();
```
Comment (case141) [inline src/main/java/seedu/address/model/source/Source.java]: Why is a set necessary? According to what was discussed before, shouldn’t a source have only a single bibliography?

**Review by case141 — DISMISSED** [review]

File: src/main/java/seedu/address/model/source/Source.java (hunk 1 above)
Comment (DoItTomorrow) [reply src/main/java/seedu/address/model/source/Source.java]: Each biblioField represents one field, such as "Name of website" etc as shown in MS word dialogue box below. I implemented it this way to match tag implementation. That way add, search and edit commands can use existing code. I can always just implement it as a single string array, but our whole project is implemented it this way, with discrete objects for everything.

image

**Comment by DoItTomorrow** [comment]
> Can you please specify what commands are affected by this update? For example I now see `search` would need to be able to take in and search by an `author` argument.

Author is implemented it the same way as Title. As it stands, your existing command would not search for anything in the Author field but it should still work fine. If you want to include it your search, you can reuse the code you use for other fields.

BiblioFields are implemented similar to Tags, except each BiblioField contains 2 strings, .type and .body. You can similar reuse any code for Tag searching for searching through BiblioFields.

**Comment by suriruhani** [comment]
> > Can you please specify what commands are affected by this update? For example I now see `search` would need to be able to take in and search by an `author` argument.
> 
> Author is implemented it the same way as Title. As it stands, your existing command would not search for anything in the Author field but it should still work fine. If you want to include it your search, you can reuse the code you use for other fields.
> 
> BiblioFields are implemented similar to Tags, except each BiblioField contains 2 strings, .type and .body. You can similar reuse any code for Tag searching for searching through BiblioFields.

Ok, noted. let us merge this soon so the affected commands have time to incorporate the changes.

File: src/main/java/seedu/address/model/source/Source.java (hunk 1 above)
Comment (DoItTomorrow) [reply src/main/java/seedu/address/model/source/Source.java]: Upon clarification, the issue seems to be that a Source can have multiple BiblioFields of the same type, which should not be the case. I'll resolve that by tonight.

**Review by ToonDragon — APPROVED** [review]
Added bliblo fields and author class.  Does not touch the existing implementation since there is still no real interaction with the model itself.

**Pull request merged**